pkg/channelmetrics/observablechan.go

```diff
@@ -64,21 +65,22 @@ func (oc *ObservableChan[T]) Send(ctx context.Context, item T) {
 
 // Recv receives an item from the channel and records the duration taken to do so.
 // It also updates the current size of the channel buffer.
-func (oc *ObservableChan[T]) Recv(_ context.Context) T {
-	startTime := time.Now()
-	defer func() {
-		oc.metrics.RecordConsumeDuration(time.Since(startTime))
+func (oc *ObservableChan[T]) Recv(ctx context.Context) T {
```

Comment:
**Suggestion (optional):**

I'm thinking two methods will cover potential use cases:

```go
// also returns an error in case of cancellation
func (oc *ObeservableChan[T]) RecvCtx(ctx context.Context) (T, error) {
    // ...
}

func (oc *ObservableChan[T]) Recv() T {
    // blocks forever
    v, _ := oc.RecvCtx(context.Background())
    return v
}
```

Reply:
Same goes for `Send`.

Reply:
Just to confirm, are you proposing adding two additional methods so it can be used in both blocking and non-blocking fashions?

Reply:
Yeah, well both blocking. The `RecvCtx` and `SendCtx` would also return an `error` so the caller knows whether it returned because the context was cancelled.